modin/data_management/functions/groupby_function.py

```diff
@@ -86,14 +87,10 @@ def map(
                     axis=1,
                 )
                 other = list(other.columns)
-            by_part = other
-        else:
-            by_part = by
+            other = wrap_into_list(other, groupby_args.pop("level", []))
```

Comment:
backend obtains two parameters that specify on what we're grouping on:
(this spliting happens [here](https://github.com/modin-project/modin/pull/2790/files#diff-986e3203c7e50cfc6e7ae32916c5c900c9ed5ff1f3b5cb7d45fca4cd4969e41eR407-R410) on the API layer)
1. `by` - column labels
2. `groupby_args["level"]` - index level labels

since pandas backend implements groupby via pandas, we must follow its rules of passing `by`, either we're moving everything in `by`, or keep only `level` parameter.

If `by` is None, we're keeping everything as is, so only `level` parameter will be passed into pandas groupby.
If `by` is specified, we're moving `level` content to `by`, so only `by` parameter will be passed into pandas groupby.

Reply:
Why not keep both? if `by is None` then `level is not None`. Each of the workers has the index, so there should be no conflict here.

Reply:
We keep both if `by is None`:
>If by is None, we're keeping everything as is...

This selected line is for the `by is not None` case and when we need to move `level` to `by`.